cogs/puzzle.py

```diff
@@ -19,33 +26,43 @@ def __init__(self, bot):
     async def submit_answer(
         self, interaction: discord.Interaction, puzzle_id: str, answer: str
     ):
-        puzzle = await find_puzzle(puzzle_id)
-        if not puzzle or not can_access_puzzle(puzzle, interaction.user.id):
-            return await interaction.response.send_message(
+        await interaction.response.defer()
+        puzzle = await get_puzzle(puzzle_id)
+        player = await get_player(str(interaction.user.id))
+        if not puzzle or not await can_access_puzzle(puzzle, player.team_name):
+            return await interaction.followup.send(
                 "No puzzle with the corresponding id exist!"
```

Comment:
Sorry, just grammatical nitpicks but it should be "No puzzle with the corresponding ID exists!"

Reply:
thank - good catch!